driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/crypt/Crypt.java

```diff
@@ -173,7 +178,9 @@ public void close() {
         if (commandMarker != null) {
             commandMarker.close();
         }
-        keyRetriever.close();
+        if (internalClient != null) {
+            internalClient.close();
```

Comment:
Do we want to to consider utilizing the try-with-resources block when closing multiple resources?

1. This way nulls are handled for us ([_"A resource is closed only if it initialized to a non-null value"_](https://docs.oracle.com/javase/specs/jls/se15/html/jls-14.html#jls-14.20.3)).
2. If the `close` method of any resource throws a `Throwable`, other resources are still closed, and none of the throwables are lost: [_"An exception from the closing of one resource does not prevent the closing of other resources. Such an exception is suppressed if an exception was thrown previously by an initializer, the try block, or the closing of a resource."_](https://docs.oracle.com/javase/specs/jls/se15/html/jls-14.html#jls-14.20.3) ("suppressed" means [`Throwable.addSuppressed`](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#addSuppressed-java.lang.Throwable-), such exceptions are shown in stack trances).

Here is an example:
```java
@Override
public void close() {
    try (final AutoCloseable _4 = keyManagementService;
        final AutoCloseable _3 = internalClient;
        final AutoCloseable _2 = commandMarker;
        final AutoCloseable _1 = mongoCrypt) {//closed in the order _1, ..., _4
    } catch (final Exception e) {//this is needed because AutoCloseable.close() throws Exception
        throw new RuntimeException(e);
    }
}
```
However, this does require marking the relevant classes as `implements AutoCloseable`, which is not a backward compatible change exactly because its `close` method is declared as `throws Exception`.

The problem with the current approach is that failing `mongoCrypt.close()` results in not closing any of the rest of the resources.

Reply:
Ah, `KeyManagementService` is internal, so can be easily made `Autocloseable` (https://github.com/mongodb/mongo-java-driver/pull/630/commits/37b25ff7f674cd0209ed9993bfcf11bb042c7e99#diff-81695b3bfe7a24372a999b1d55f9af8799b05e515d88011e14e6332f6b6ed58dR43). 👍

Reply:
Good catch @stIncMale

Reply:
I like this approach, though Intellij warns about unused local variables.   We don't have to worry about catching exceptions because all of our close() method overrides are define to not throw checked exceptions.

Done